pennylane/workflow/qnode.py

```diff
@@ -110,15 +110,33 @@ def _to_qfunc_output_type(
     if has_partitioned_shots:
         return tuple(_to_qfunc_output_type(r, qfunc_output, False) for r in results)
 
+    qfunc_output_leaves, qfunc_output_structure = qml.pytrees.flatten(
+        qfunc_output, is_leaf=lambda obj: isinstance(obj, (qml.measurements.MeasurementProcess))
+    )
+    num_of_measurements = len(qfunc_output_leaves)

```

Comment:
Minor stylistic preference, take it or leave it: since the code that creates these variables is just about as simple as the variable names, and they are only used one place, I would suggest getting rid of `num_of_measurements` (and `num_of_results`) and just comparing the two lengths directly in line 128. 

Fewer variables to keep track of in the code makes it faster for me to read 🤷‍♀️

Reply:
Agreed! 😸